.gitignore

```diff
@@ -2,6 +2,7 @@ target
 .*.swp
 *.jar
 .vscode
+.idea
```

Comment:
Normally I would reject this btw, but it's right after a line of `.vscode` so .. uh, guess we accepted that a while ago.

Reply:
What's the problem with it? It just helps contributors not accidentally commit unwanted files.

Reply:
It tends to accumulate a list of editor-specific files over time, just because one more person used a different setup.
These should go into one's global gitignore (IMO); but again: I didn't reject it here and it's fine.

Reply:
I always forget that global `.gitignore` is a thing :)

(though I wonder if it's all that practical, there are projects where I've wanted to check in editor configuration though nowadays I prefer a `contrib/ide` directory instead)